src/service_client/assisted_service_api.py

```diff
@@ -47,10 +76,16 @@ def get_host(self, configs: Configuration) -> str:
         return parsed_host._replace(netloc=parsed_inventory_url.netloc, scheme=parsed_inventory_url.scheme).geturl()
 
     @classmethod
-    def set_config_auth(cls, c: Configuration, offline_token: Optional[str]) -> None:
-        if not offline_token:
-            log.info("OFFLINE_TOKEN not set, skipping authentication headers")
-            return
+    def set_config_auth(
+        cls, c: Configuration, offline_token: Optional[str], service_account: Optional[ServiceAccount]
+    ) -> None:
+        authentication_method = AuthenticationMethod.SERVICE_ACCOUNT
+        if service_account is None or not service_account.is_provided():
```

Comment:
This is a bit confusing, setting the method as SERIVCE_ACCOUNT before validating the method type